### PR TITLE
Tracing: Make span name to be egress host_header, e.g., egress api

### PR DESCRIPTION
**Pull request by RomanDzhabarov**
@lyft/network-team

### CODE REVIEW

File: source/common/tracing/http_tracer_impl.cc [hunk 1]
```diff
@@ -135,8 +135,13 @@ HttpTracerImpl::HttpTracerImpl(DriverPtr&& driver, const LocalInfo::LocalInfo& l
 
 SpanPtr HttpTracerImpl::startSpan(const Config& config, Http::HeaderMap& request_headers,
                                   const Http::AccessLog::RequestInfo& request_info) {
+  std::string operation_name = config.operationName();
+  if (request_headers.Host()) {
```
Comment (mattklein123) [inline source/common/tracing/http_tracer_impl.cc]: requests must have host header.

File: test/common/tracing/http_tracer_impl_test.cc [hunk 2]
```diff
@@ -475,6 +475,28 @@ TEST(HttpTracerImplTest, BasicFunctionalityNodeSet) {
   tracer.startSpan(config, request_headers, request_info);
 }
 
+TEST(HttpTracerImplTest, TestOperationName) {
+  LocalInfo::MockLocalInfo local_info;
+  SystemTime time;
+  Http::AccessLog::MockRequestInfo request_info;
+  EXPECT_CALL(request_info, startTime()).WillOnce(Return(time));
```
Comment (mattklein123) [inline test/common/tracing/http_tracer_impl_test.cc]: This should be part of the default mock via ON_CALL, can you add this to default mock and clean this up elsewhere in this file.

File: source/common/tracing/http_tracer_impl.cc (hunk 1 above)
Comment (RomanDzhabarov) [inline source/common/tracing/http_tracer_impl.cc]: ah, right, it's checked in the same decode headers

File: source/common/tracing/http_tracer_impl.cc [hunk 3]
```diff
@@ -135,8 +135,11 @@ HttpTracerImpl::HttpTracerImpl(DriverPtr&& driver, const LocalInfo::LocalInfo& l
 
 SpanPtr HttpTracerImpl::startSpan(const Config& config, Http::HeaderMap& request_headers,
                                   const Http::AccessLog::RequestInfo& request_info) {
+  std::string operation_name =
```
Comment (mattklein123) [inline source/common/tracing/http_tracer_impl.cc]: fmt::format is kind of slow. Can we just do direct string operations here.

File: test/common/tracing/http_tracer_impl_test.cc [hunk 4]
```diff
@@ -429,50 +430,47 @@ TEST(HttpNullTracerTest, BasicFunctionality) {
...
-  Http::AccessLog::MockRequestInfo request_info;
-  EXPECT_CALL(request_info, startTime()).WillOnce(Return(time));
-
-  MockConfig config;
-  const std::string operation_name = "operation";
-  EXPECT_CALL(config, operationName()).WillOnce(ReturnRef(operation_name));
+class HttpTracerImplTest : public Test {
+public:
+  HttpTracerImplTest() {
+    ON_CALL(request_info_, startTime()).WillByDefault(Return(start_time_));
```
Comment (mattklein123) [inline test/common/tracing/http_tracer_impl_test.cc]: This should go directly in the request_info_ mock. It doesn't need to be here.

File: source/common/tracing/http_tracer_impl.cc (hunk 3 above)
Comment (RomanDzhabarov) [inline source/common/tracing/http_tracer_impl.cc]: done

File: test/mocks/tracing/mocks.h [hunk 5]
```diff
@@ -15,6 +15,8 @@ class MockConfig : public Config {
   ~MockConfig();
 
   MOCK_CONST_METHOD0(operationName, const std::string&());
+
+  const std::string operation_name_ = "operation";
```
Comment (mattklein123) [inline test/mocks/tracing/mocks.h]: Someone might want to change this. I would make this non-const.

**Pull request merged**